### ISSUE TITLE
[BUG] With Kubevela v1.1.1  cluster.core.oam.dev/v1alpha1  does not repond when multicluster is not enabled

### ISSUE THREAD

**Issue by jriguera** [kubevela/kubevela#2319]
**Describe the bug**
After installing Kubevela V1.1.1 with the official helm chart and with apiserver enabled (now by default) we started to see errors from kubectl `error: unable to retrieve the complete list of server APIs: cluster.core.oam.dev/v1alpha1: the server is currently unable to handle the request`. 

It seems the problem comes from https://github.com/oam-dev/kubevela/blob/master/charts/vela-core/templates/cluster-gateway.yaml#L54-L84 . The APIService makes reference to a service without backend, because the deployment of the cluster gateway is not triggered (multicluster is disabled by default).

For example, deleting a namespace in the cluster will block the operation and set the status to `terminating`:
```
kubectl get namespace jose -o json
{
    "apiVersion": "v1",
    "kind": "Namespace",
    "metadata": {
        "creationTimestamp": "2021-03-03T14:07:46Z",
        "deletionTimestamp": "2021-09-17T08:29:03Z",
        "labels": {
            "kubernetes.io/metadata.name": "jose"
        },
        "name": "jose",
        "resourceVersion": "514073371",
        "uid": "100eeabd-c3a5-4f81-88cf-8cda9193e880"
    },
    "spec": {
        "finalizers": [
            "kubernetes"
        ]
    },
    "status": {
        "conditions": [
            {
                "lastTransitionTime": "2021-09-17T08:29:08Z",
                "message": "Discovery failed for some groups, 1 failing: unable to retrieve the complete list of server APIs: cluster.core.oam.dev/v1alpha1: the server is currently unable to handle the request",
                "reason": "DiscoveryFailed",
                "status": "True",
                "type": "NamespaceDeletionDiscoveryFailure"
            },
            {
                "lastTransitionTime": "2021-09-17T08:29:10Z",
                "message": "All legacy kube types successfully parsed",
                "reason": "ParsedGroupVersions",
                "status": "False",
                "type": "NamespaceDeletionGroupVersionParsingFailure"
            },
            {
                "lastTransitionTime": "2021-09-17T08:29:10Z",
                "message": "All content successfully deleted, may be waiting on finalization",
                "reason": "ContentDeleted",
                "status": "False",
                "type": "NamespaceDeletionContentFailure"
            },
            {
                "lastTransitionTime": "2021-09-17T08:29:10Z",
                "message": "All content successfully removed",
                "reason": "ContentRemoved",
                "status": "False",
                "type": "NamespaceContentRemaining"
            },
            {
                "lastTransitionTime": "2021-09-17T08:29:10Z",
                "message": "All content-preserving finalizers finished",
                "reason": "ContentHasNoFinalizers",
                "status": "False",
                "type": "NamespaceFinalizersRemaining"
            }
        ],
        "phase": "Terminating"
    }
}
```

**To Reproduce**
1. Create a namespace in the cluster (empty)
2. Update KubeVela  to v1.1.1 (official helm chat) and make sure apiserver is enabled and multicluster is disabled. Those values are defaults in v1.1.1,
3. Delete the namespace created in the previous step
4. See error: `Discovery failed for some groups, 1 failing: unable to retrieve the complete list of server APIs: cluster.core.oam.dev/v1alpha1: the server is currently unable to handle the request

**Expected behavior**
Namespace deleted without errors.

**KubeVela Version**
Kubevela: 1.1.1
Client: 1.1.1  (linux-amd64)

**Cluster information**
Kubernetes: v1.21.4-gke.301
Kubectl: v1.21.5

**Additional context**
In this example, the namespace has nothing to do with KubeVela resources/apps.

**Comment by wonderflow** [comment]
Sorry, this is can be the image is in wrong tag as we didn't push the v1.1.1 image tag, please retry as we have push the image:

`oamdev/cluster-gateway:v1.1.1`

**Comment by Somefive** [comment]
The problem is caused by a mistake in cluster-gateway yaml file since the condition section should also includes the service and the APIService. Thanks for the report.

**Comment by rtrive** [comment]
I'm using vela-core 1.2.2 but this problem is still here for me. The error that I get is `couldn't get resource list for cluster.core.oam.dev/v1alpha1: the server is currently unable to handle the request`

**Comment by wonderflow** [comment]
ping @Somefive

**Comment by rtrive** [comment]
apologise @wonderflow but I made a mistake. This issue is related to `multicluster disabled`, instead I have `multicluster enabled`. Do you prefer that I'll open a new issue?

**Comment by wonderflow** [comment]
@rtrive yes, let's create a new issue for this. I think we need some more background information about your issue.
